projects/prison-case-notes-to-probation/src/main/kotlin/uk/gov/justice/digital/hmpps/integrations/delius/service/DeliusService.kt

```diff
@@ -24,21 +27,50 @@ class DeliusService(
     private val caseNoteTypeRepository: CaseNoteTypeRepository,
     private val offenderRepository: OffenderRepository,
     private val assignmentService: AssignmentService,
-    private val relatedService: CaseNoteRelatedService
+    private val relatedService: CaseNoteRelatedService,
+    private val telemetryService: TelemetryService,
+    @Value("\${ignore.message}")
+    private val ignoreMessage: String?,
 ) : AuditableService(auditedInteractionService) {
     @Transactional
     fun mergeCaseNote(@Valid caseNote: DeliusCaseNote) = audit(CASE_NOTES_MERGE) {
         it["nomisId"] = caseNote.header.noteId
 
         val existing = caseNoteRepository.findByNomisId(caseNote.header.noteId)
 
-        val entity = if (existing == null) caseNote.newEntity() else existing.updateFrom(caseNote)
+        val entity = if (existing == null) checkOCGMessages(caseNote) else existing.updateFrom(caseNote)
```

Comment:
Will this only apply to new case notes?  Do we also want to ignore any updates that contain OCG information?

Also this would be better placed in the PrisonCaseNoteFilters class, there are some existing predicates in there that we use to filter out case notes.  That would also handle the logging to Telemetry for you.

Reply:
This will only apply to new case notes.  I did consider adding at the handler layer, but happy where it is at, unless you would prefer it t be moved.

Reply:
done